### PR TITLE
hyper-v: add support for EnhancedSessionTransportType. 

### PR DESCRIPTION
**Pull request by camjjack**
Addresses #9823

First contribution to this project so keen to find out if I've missed anything else that I need to do.

### CODE REVIEW

**Comment by zeenlym** [comment]
Does anyone plan to review this pull request ?

File: plugins/providers/hyperv/action/configure.rb [hunk 1]
```diff
@@ -97,6 +97,11 @@ def call(env)
               env[:machine].provider_config.vm_integration_services)
           end
 
+          if env[:machine].provider_config.enable_enhanced_session_mode
```
Comment (soapy1) [inline plugins/providers/hyperv/action/configure.rb:100]: This looks great, just one more thing. Users should be able to disable EnhancedSessionTransportType as well as enable it. 
So something like
```
$ cat Vagrantfile
# Vagrantfile
 ...
    h.vm.provider "hyperv" do |b|
      b.enable_enhanced_session_mode = true
    end
 ...

$ vagrant up box
```
will bring up the box with the EnhancedSessionTransportType set to `HvSocket`. Then editing the Vagrantfile setting enable_enhanced_session_mode to false 
```
$ cat Vagrantfile
# Vagrantfile
 ...
    h.vm.provider "hyperv" do |b|
      b.enable_enhanced_session_mode = false
    end
  ...

$ vagrant reload box
```
should reload the box with the EnhancedSessionTransportType set to `VMBus`. However, currently it leaves the EnhancedSessionTransportType set to `HvSocket`

Comment (camjjack) [reply plugins/providers/hyperv/action/configure.rb:100]: Thanks for the feedback. Added this in commit. 1ed4c882dc6adc19b330da146a432265cd152a34
Note as its a bool setting to false is the same as not present, which is I think the best behavior for a new feature.

Comment (soapy1) [reply plugins/providers/hyperv/action/configure.rb:100]: Awesome, this looks great! Thanks 🙏

**Comment by camjjack** [comment]
Is anyone able to action this pull request? Its been approved for a while but its becoming stale because it hasn't been merged.

**Pull request merged**